_source/_posts/2022-09-26-java-jetty-oauth.md

```diff
@@ -0,0 +1,1017 @@
+
+The two Java servlet classes are the same in both projects. I did find that Gretty and Gradle crashed unless I added an empty `src/main/webapp` directory (this wasn't necessary with Maven).
+
+The `build.gradle` file is analogous to the `pom.xml` file.
+
+```bash
+plugins {
+    id 'java'
+    id 'war'
+    id 'org.gretty' version '4.0.3'
+}
+
+repositories {
+    jcenter()
+}
+
+dependencies {
+    providedCompile 'jakarta.servlet:jakarta.servlet-api:6.0.0'
+}
+
+gretty {
+    contextPath = '/'
+}
+```
+
+This is about as simple as it gets. The same Jakarta-based servlet API is included as a provided dependency, and the Gretty plugin is included in the `plugins` block.
+
+The context path is configured to be the root path. If you want to see the other configuration properties, take a look at [the Gretty configuration docs](https://gretty-gradle-plugin.github.io/gretty-doc/Gretty-configuration.html).
+
+You can also look at the Gradle tasks that the plugin adds [in the docs](https://gretty-gradle-plugin.github.io/gretty-doc/Gretty-tasks.html).
+
+To run the web app, use the `appRun` task.
+
+```bash
+./gradlew appRun
+```
+
+You can test the two endpoints: `/hikes` and `/ahoy`.
+
+```bash
+$ http :8080/hikes
+HTTP/1.1 200 OK
+...
+
+Wonderland Trail
+South Maroon Peak
+Tour du Mont Blanc
+Teton Crest Trail
+Everest Base Camp via Cho La Pass
+Kesugi Ridge
+
+$: http :8080/ahoy
+HTTP/1.1 200 OK
+...
+
+Ahoy!
+```
+
+You've seen how to build a web service using Jetty in both Gradle and Maven. Next, you'll build one using Spring Boot and Maven.
+
+## Build a web service with Spring Boot and Jetty
+
+By default, Spring Boot is configured to use an embedded Tomcat server. However, this is easily configurable. The [Spring Boot docs on embedded web servers](https://docs.spring.io/spring-boot/docs/2.0.6.RELEASE/reference/html/howto-embedded-web-servers.html) is a good resource. To switch to Jetty, you need to do two things:
+
+1. disable the default embedded Tomcat server
+2. enable an embedded Jetty server
+
+This is done in the `pom.xml` file by these two dependency configuration blocks. The first removes the Tomcat dependency from `spring-boot-starter-web-no-auth`, and the second adds the `spring-boot-starter-jetty` dependency.
+
+```xml
+<dependency>
+    <groupId>org.springframework.boot</groupId>
+    <artifactId>spring-boot-starter-web</artifactId>
+    <exclusions>
+        <!-- Exclude the Tomcat dependency -->
+        <exclusion>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter-tomcat</artifactId>
+        </exclusion>
+    </exclusions>
+</dependency>
+<!-- Use Jetty instead -->
+<dependency>
+    <groupId>org.springframework.boot</groupId>
+    <artifactId>spring-boot-starter-jetty</artifactId>
+</dependency>
+```
+
+Here is the full `pom.xml`. Notice the Spring Boot version is 2.7.4.
```

Comment:
@moksamedia Do you have a repo you created this example in? If so, can you please upgrade it to 2.7.4 and I'll QA against it?

Reply:
Found it! https://github.com/moksamedia/okta-spring-boot-jetty-2022